config/src/test/java/org/springframework/security/config/authentication/JdbcUserServiceBeanDefinitionParserTests.java

```diff
@@ -160,6 +162,22 @@ public void rolePrefixIsUsedWhenSet() {
 		assertThat(AuthorityUtils.authorityListToSet(rod.getAuthorities())).contains("PREFIX_ROLE_SUPERVISOR");
 	}
 
+	@Test
+	public void testEmptyDataSourceRef() {
+		String xml = "<authentication-manager>"
+				+ "  <authentication-provider>"
+				+ "    <jdbc-user-service data-source-ref=''/>"
```

Comment:
What happens when the data-source-ref attribute is missing entirely? Please include a test for that as well.

Reply:
Done!